fbw-a32nx/src/systems/instruments/src/EFB/Failures/FailureGenerators/AltitudeFailureGenerator.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useMemo } from 'react';
+import { useSimVar } from '@instruments/common/simVars';
+import {
+    activateRandomFailure, basicData, FailureGenContext, FailureGenData, failureGeneratorCommonFunction,
+    FailurePhases, flatten, setNewSetting,
+} from 'instruments/src/EFB/Failures/FailureGenerators/RandomFailureGen';
+import { usePersistentProperty } from '@instruments/common/persistence';
+import { t } from 'instruments/src/EFB/translation';
+import { findGeneratorFailures } from 'instruments/src/EFB/Failures/FailureGenerators/FailureSelection';
+import { ArrowDownRight, ArrowUpRight } from 'react-bootstrap-icons';
+import { ButtonIcon, FailureGeneratorChoiceSetting, FailureGeneratorSingleSetting } from 'instruments/src/EFB/Failures/FailureGenerators/FailureGeneratorSettingsUI';
+
+const settingName = 'EFB_FAILURE_GENERATOR_SETTING_ALTITUDE';
+const additionalSetting = [2, 1, 0, 80, 250];
+const numberOfSettingsPerGenerator = 5;
+const uniqueGenPrefix = 'A';
+const failureGeneratorArmed :boolean[] = [];
+const genName = 'Altitude';
+const alias = () => t('Failures.Generators.GenAlt');
+const disableTakeOffRearm = false;
+const rolledDice:number[] = [];
+
+export const failureGenConfigAltitude : ()=>FailureGenData = () => {
+    const [setting, setSetting] = usePersistentProperty(settingName);
+    const settings = useMemo(() => {
+        const splitString = setting?.split(',');
+        if (splitString) return splitString.map(((it : string) => parseFloat(it)));
+        return [];
+    }, [setting]);
+    return {
+        setting,
+        setSetting,
+        settings,
+        numberOfSettingsPerGenerator,
+        uniqueGenPrefix,
+        additionalSetting,
+        onErase,
+        failureGeneratorArmed,
+        genName,
+        alias,
+        disableTakeOffRearm,
+        generatorSettingComponents,
```

Comment:
setting was indeed a remnant code, I removed it from the structrure to keep "settings" only.
I thought the memo was necessary as I don't want a string parsing to happen at each loop. This is really cpu heavy usually.
Can I keep that memo ?

Reply:
If there's a big performance impact then its ok, if in practice its not a big deal then it would be better to not affect the legibility of the code, but otherwise its fine.

Reducing the size of the object in any case would be a good optimisation. (i,e. no ``setting`` + ``settings`` in properties)

Reply:
Thanks for the explanation, topic resolved then.